ipc_receivefile.c

```diff
@@ -0,0 +1,402 @@
+////////////////////////////////////////////////////////////////////////////////
+//IPC RECEIVER
+//
+//
+//
+////////////////////////////////////////////////////////////////////////////////
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <sys/iofunc.h>
+#include <sys/dispatch.h>
+#include <sys/neutrino.h>
+
+#include "ipc_common.h"
+
+#include <getopt.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+#include <sys/mman.h>
+#include <process.h>
+#include <string.h>
+#include <unistd.h>
+
+typedef union
+{
+	uint16_t msg_type;
+	struct _pulse pulse;
+	cksum_header_t cksum_hdr;
+} msg_buf_t;
+
+typedef union
+{
+	uint16_t type;
+	struct _pulse pulse;
+	get_shmem_msg_t get_shmem;
+	changed_shmem_msg_t changed_shmem;
+} recv_buf_t;
+
+int getFileSize(int field_descriptor);

```

Comment:
You need to remove the declaration of getFileSize because the compiler think it is defined in this .c instead of ipc_common.c

Reply:
The declaration of getFileSize in ipc_receive and ipc_send has been removed. Please check it again.
I have modified a lot of structure things after expiration of the license, so please tell me if you find more problems.